test/server/http/health_check_test.cc

```diff
@@ -49,6 +54,30 @@ class HealthCheckFilterTest : public testing::Test {
   NiceMock<Http::MockStreamDecoderFilterCallbacks> callbacks_;
   Http::TestHeaderMapImpl request_headers_;
   Http::TestHeaderMapImpl request_headers_no_hc_;
+
+  class MockHealthCheckCluster : public Upstream::MockCluster {
```

Comment:
Rather than creating a one-off mock here, I think you should add a helper function in the test file that just uses the existing MockCluster. I think something like this will work to create values you can use for cluster_www1/2:

```
MockCluster makeHealthCheckCluster(uint64_t membership_total, uint64_t membership_healthy) {
  MockCluster cluster;
  cluster.info_->stats_.membership_total_.set(membership_total);
  cluster.info_->stats_.membership_healthy_.set(membership_healthy);
  return cluster;
}
```

Reply:
Thanks for the pointer! `MockCluster` isn't copyable (something in the gmock macros seems to be explicitly deleting the copy constructor), but I was able to use your approach to replace all of `MockHealthCheckCluster` with just a constructor that sets the stats in `MockCluster`.